pkg/cstor/volume/targetserver/server.go

```diff
@@ -0,0 +1,174 @@
+/*
+Copyright 2019 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package targetserver
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+
+	"k8s.io/klog"
+
+	cstorv1alpha1 "github.com/openebs/maya/pkg/cstor/volume/v1alpha1"
+	env "github.com/openebs/maya/pkg/env/v1alpha1"
+	"github.com/pkg/errors"
+)
+
+var (
+	endOfLine          = "\r\n"
+	respOk             = "Ok"
+	respErr            = "Err"
+	volumeMgmtUnixSock = "/var/run/volume_mgmt_sock"
+)
+
+// Reader reads the data from wire untill error or endOfLine occurs
+func Reader(r io.Reader) (string, error) {
+	req := []string{}
+	//collect bytes into fulllines buffer till the end of line character is reached
+	completeBytes := []byte{}
+	for {
+		buf := make([]byte, 1024)
+		n, err := r.Read(buf[:])
+		if err != nil {
+			if err == io.EOF {
+				klog.Info("Reached End Of file")
+				break
+			}
+			return "", errors.Wrapf(err, "failed to read data on wire")
+		}
+		if n > 0 {
+			completeBytes = append(completeBytes, buf[0:n]...)
+			if strings.HasSuffix(string(completeBytes), endOfLine) {
+				lines := strings.Split(string(completeBytes), endOfLine)
```

Comment:
why to `Split` and why to `append`?

Reply:
This piece of code needs fix.. please put the communication protocol in comments